Italian-Docs

justoverclockl opened the pull request:
Italian translation for flarum docs.

askvortsov1 reviewed (changes requested):
Thank you for the PR! These should be added as separate files with a navigation configuration similar to that of the chinese and turkish translations we have now.

justoverclockl commented:
can u explain me step by step what i have to do? i'm really new to github :)

u mean that i have to create a folder [it]?

Littlegolden commented:
@justoverclockl I hope this tutorial can help you~ https://discuss.flarum.org.cn/d/2433

justoverclockl commented:
> @justoverclockl I hope this tutorial can help you~ https://discuss.flarum.org.cn/d/2433

mmmhhh not really, i need to install vuepress locally on my computer?

Littlegolden commented:
If you don't need to preview the docs website locally, you don't need to run `npm install`, just config the locale settings and create the folder where the Italian documentation will be stored.

justoverclockl commented:
> If you don't need to preview the docs website locally, you don't need to run `npm install`, just config the locale settings and create the folder where the Italian documentation will be stored.

can u check now, now the folder is updated with original structure, and IT folder with translation

Littlegolden commented:
yes it's right now. and don't forget the config files in `/.vuepress/config`

justoverclockl commented:
> yes it's right now. and don't forget the config files in `/.vuepress/config`

done!

Littlegolden commented on docs/.vuepress/config/locales/it/nav.js:
```diff
@@ -0,0 +1,13 @@
+module.exports = [
+    { text: 'Guide', link: '/' },
```
the link shoule be `/it/`

justoverclockl replied:
done!

Littlegolden commented on docs/.vuepress/config/locales/it/nav.js:
```diff
@@ -0,0 +1,13 @@
+module.exports = [
+    { text: 'Guide', link: '/' },
+    { text: 'Extend', link: '/extend/' },
```
you can keep extend docs link `/extent` before you translated extend docs, this will link readers to the English document

Littlegolden commented on docs/.vuepress/config/locales/it/sidebar.js:
```diff
@@ -0,0 +1,104 @@
+    // }
+  ],
+
+  '/': [
```
need change to `/it/`

justoverclockl replied:
done!

askvortsov1 reviewed (changes requested):
Overall, looking quite good! A few more changes needed:

- `'it'` needs to be added to `.vuepress/index.js`'s `lang` array so that Italian shows up in the top right "select language" dropdown
- For extend docs that haven't yet been translated, instead of copy-pasting the English version, you should put a short message saying that they haven't been translated, and to refer to the English version. We want to avoid maintaining multiple copies of translations in the same language so that  changes can be made in one place (for that language)

justoverclockl commented:
> Overall, looking quite good! A few more changes needed:
> 
> * `'it'` needs to be added to `.vuepress/index.js`'s `lang` array so that Italian shows up in the top right "select language" dropdown
> * For extend docs that haven't yet been translated, instead of copy-pasting the English version, you should put a short message saying that they haven't been translated, and to refer to the English version. We want to avoid maintaining multiple copies of translations in the same language so that  changes can be made in one place (for that language)

vuepress/index.js added.

i'm new to github so i do not understand well what i have to do with the Extend folder (actually i'm working on it locally)

askvortsov1 commented on docs/.vuepress/config/locales/index.js:
```diff
@@ -1,5 +1,5 @@
-const langs = ['en', 'zh', 'tr'];
-const paths = { en: '/' };
+const langs = ['en', 'zh', 'tr','it'];
+const paths = { it: '/it/' };
```
This should be restored to how it was before

Littlegolden replied:
In fact, the path does not need to be modified

askvortsov1 replied:
`langs` should stay as it is now (with the `it` added), but `paths` should be changed back; this configures it so that the english docs will show up at `docs.flarum.org` instead of `docs.flarum.org/en` since those are the default

justoverclockl replied:
oh, sorry for that, done!

askvortsov1 reviewed (changes requested):
Ah, sorry I see now that you're working through translations for the extend stuff; that's fine then, we'll just wait till they're done to merge this.

One thing we'll want to add: since the dev team writes docs in English, translated docs might get out of date. We should add a notice about this similar to how the Chinese docs do (https://docs.flarum.org/zh/, for instance)

justoverclockl commented:
> Overall, looking quite good! A few more changes needed:
> 
> * `'it'` needs to be added to `.vuepress/index.js`'s `lang` array so that Italian shows up in the top right "select language" dropdown
> * For extend docs that haven't yet been translated, instead of copy-pasting the English version, you should put a short message saying that they haven't been translated, and to refer to the English version. We want to avoid maintaining multiple copies of translations in the same language so that  changes can be made in one place (for that language)

ok, can u tell me where this need to be applied? and maybe an example of the text? on that links i see only chinese words :)

Littlegolden commented:
The notice component is [here](https://github.com/flarum/docs/tree/master/docs/.vuepress/components)，take a look at how the Chinese docs use it. https://github.com/flarum/docs/blame/master/docs/zh/install.md#L1-L3

justoverclockl commented:
> The notice component is [here](https://github.com/flarum/docs/tree/master/docs/.vuepress/components)，take a look at how the Chinese docs use it: https://github.com/flarum/docs/blame/master/docs/zh/install.md#L1-L3
> 
> You can just use the component for now. I believe we can use js to insert notice text in different languages which I can do that in the daytime tomorrow.

done (not sure, but done) :)

askvortsov1 commented:
> > The notice component is [here](https://github.com/flarum/docs/tree/master/docs/.vuepress/components)，take a look at how the Chinese docs use it: https://github.com/flarum/docs/blame/master/docs/zh/install.md#L1-L3
> > You can just use the component for now. I believe we can use js to insert notice text in different languages which I can do that in the daytime tomorrow.
> 
> done (not sure, but done) :)

Not quite. You need to copy that file (outdated from components) to a new one with `it` in the name, replace the Chinese text with italian equivalent, and reference that new component at the top of every time

matteocontrini commented on docs/it/README.md:
```diff
@@ -0,0 +1,17 @@
+
+* **Bello e responsive.**  Il suo design si adatta a dispositivi fissi, mobili e tablet in maniera responsive. Flarum è progettato con cura per essere coerente e intuitivo su tutte le piattaforme..
+
+* **Potente ed estendibile.** Personalizza, estendi e integra Flarum per creare la tua community unica. L'architettura di Flarum è molto flessibile grazie alle sue [powerful Extension API](/extend/).
```
```suggestion
* **Potente ed estendibile.** Personalizza, estendi e integra Flarum per creare la tua community unica. L'architettura di Flarum è molto flessibile grazie alle sue [potenti API per le estensioni](/extend/).
```

matteocontrini commented on docs/it/README.md:
```diff
@@ -0,0 +1,17 @@
+
+* **Potente ed estendibile.** Personalizza, estendi e integra Flarum per creare la tua community unica. L'architettura di Flarum è molto flessibile grazie alle sue [powerful Extension API](/extend/).
+
+* **Gratuito e aperto.** Flarum è totalmente gratuito e rilasciato sotto [MIT license](https://github.com/flarum/flarum/blob/master/LICENSE).
```
```suggestion
* **Gratuito e aperto.** Flarum è totalmente gratuito e rilasciato sotto [licenza MIT](https://github.com/flarum/flarum/blob/master/LICENSE).
```

matteocontrini commented on docs/it/admin.md:
```diff
@@ -0,0 +1,5 @@
+# Pannello di amministrazione
+
+Il pannello di amministrazione di Flarum un'interfaccia user-friendly per la gestione del tuo forum.
+È disponibile solo per gli utenti nel gruppo "Admin" [group](permissions.md).
```
```suggestion
È disponibile solo per gli utenti nel [gruppo](permissions.md) "Admin".
```

matteocontrini commented on docs/it/bugs.md:
```diff
@@ -0,0 +1,26 @@
+
+Hai già trovato un bug? Meraviglioso Ci piacerebbe saperlo, ma prima dovresti controllare per assicurarti di non sprecare il tuo tempo su un problema già noto:
+
+- Cerca sul [Support forum](https://discuss.flarum.org/t/support) per vedere se il bug è stato già segnalato.
```
```suggestion
- Cerca sul [forum di supporto](https://discuss.flarum.org/t/support) per vedere se il bug è stato già segnalato.
```

matteocontrini commented on docs/it/bugs.md:
```diff
@@ -0,0 +1,26 @@
+Hai già trovato un bug? Meraviglioso Ci piacerebbe saperlo, ma prima dovresti controllare per assicurarti di non sprecare il tuo tempo su un problema già noto:
+
+- Cerca sul [Support forum](https://discuss.flarum.org/t/support) per vedere se il bug è stato già segnalato.
+- Potremmo essere al lavoro su una soluzione, quindi cerca anche sul nostro [issue tracker](https://github.com/flarum/core/issues).
```
```suggestion
- Potremmo essere già al lavoro su una soluzione, quindi cerca anche sul nostro [sistema di gestione degli *issue*](https://github.com/flarum/core/issues).
```

matteocontrini commented on docs/it/bugs.md:
```diff
@@ -0,0 +1,26 @@
+- Cerca sul [Support forum](https://discuss.flarum.org/t/support) per vedere se il bug è stato già segnalato.
+- Potremmo essere al lavoro su una soluzione, quindi cerca anche sul nostro [issue tracker](https://github.com/flarum/core/issues).
+
+Se hai cercato a fondo e ne sei uscito a mani vuote, saremo lieti di ricevere il tuo rapporto. Se si tratta solo di un problema semplice (ad esempio una parola con errori di ortografia o un problema grafico), passa al paragrafo successivo. Ma se vedi errori o qualcosa è chiaramente un errore, avremo bisogno che tu raccolga prima alcune informazioni. Per favore leggi la guida alla [Troubleshooting ...
```
```suggestion
Se hai cercato a fondo e ne sei uscito a mani vuote, saremo lieti di ricevere il tuo rapporto. Se si tratta solo di un problema semplice (ad esempio una parola con errori di ortografia o un problema grafico), passa al paragrafo successivo. Ma se vedi errori o qualcosa è chiaramente un errore, avremo bisogno che tu raccolga prima alcune informazioni. Per favore leggi la guida per la [Risoluzione dei problemi](troubleshoot.md)  e seguine le istruzioni. Raccogli più informazioni possibili!
```

matteocontrini commented on docs/it/code-of-conduct.md:
```diff
@@ -0,0 +1,51 @@
+
+### Fatti sentire!
+
+Vuoi iniziare una nuova discussione? Innanzitutto, assicurati di leggere le [our FAQ](faq.md) ae segui i link per assicurarti di essere completamente informato sul progetto. Quindi trascorri un po 'di tempo a navigare nel forum, familiarizza con [il sistema dei TAG](https://discuss.flarum.org/tags), e fai qualche ricerca per parole chiave correlate al tuo argomento: ci potrebbe essere qualcuno ch ...
```
```suggestion
Vuoi iniziare una nuova discussione? Innanzitutto, assicurati di leggere le nostre [domande frequenti](faq.md) ae segui i link per assicurarti di essere completamente informato sul progetto. Quindi trascorri un po 'di tempo a navigare nel forum, familiarizza con [il sistema dei TAG](https://discuss.flarum.org/tags), e fai qualche ricerca per parole chiave correlate al tuo argomento: ci potrebbe essere qualcuno che ha già avviato una discussione simile o identica!
```

matteocontrini commented on docs/it/code-of-conduct.md:
```diff
@@ -0,0 +1,51 @@
+- Se non utilizzi un tag riservato per l'uso multilingue, pubblica solo in italiano. Non possiamo aiutarti se non possiamo leggerti!
+- Ricorda, non � necessario firmare i tuoi post. Abbiamo il tuo profilo per farci sapere chi sei.
+
+Please make the effort to help us keep things organized. Time spent tidying up is time that we can't spend getting to know you, discussing your issues, and talking about Flarum. And that, after all, is what we're all here to do!
```
Missing translation.

matteocontrini commented on docs/it/config.md:
```diff
@@ -0,0 +1,35 @@
+    'port' => '3306', // la porta di connessione, di default 3306 con MySQL
+    'strict' => false,
+  ),
+  'url' => 'https://flarum.localhost', // l' URL di installazione, vorrai cambiarlo se cambi domini
```
```suggestion
  'url' => 'https://flarum.localhost', // l'URL di installazione, vorrai cambiarlo se cambi domini
```

matteocontrini commented on docs/it/contributing.md:
```diff
@@ -0,0 +1,142 @@
+
+## Su cosa lavorare
+
+Dai un occhiata ai nostri prossimi [Traduardi (in inglese)](https://github.com/flarum/core/milestones) per una panoramica di ci� che deve essere fatto. Consulta le [Good first issue](https://github.com/flarum/core/labels/Good%20first%20issue) per un elenco di problemi con cui dovrebbe essere relativamente facile iniziare.
```
```suggestion
Dai un occhiata ai nostri prossimi [Traguardi (in inglese)](https://github.com/flarum/core/milestones) per una panoramica di ciò che deve essere fatto. Consulta le [Good first issue](https://github.com/flarum/core/labels/Good%20first%20issue) per un elenco di problemi con cui dovrebbe essere relativamente facile iniziare.
```

matteocontrini reviewed (changes requested):
Hello, I left some changes suggestions. It seems there are many other links that are untranslated, I just skimmed quickly through the first few files, but unfortunately I don't have time to review the whole thing, it's extremely long 😰

justoverclockl commented:
> Hello, I left some changes suggestions. It seems there are many other links that are untranslated, I just skimmed quickly through the first few files, but unfortunately I don't have time to review the whole thing, it's extremely long 😰

I will check again all to fix this asap!

luceos commented:
@matteocontrini and @Littlegolden thanks for reviewing, if the both of you can approve once everything is in order I think it's okay to merge 👍

matteocontrini reviewed (approved):
I'm unable to read all of it thoroughly, but I skimmed through it and pushed some fixes, especially with accents that were broken in some files. I also changed the wording of some technical terms in the Contributing page, so that the original English terms are used (e.g. "branch" shouldn't be translated).

Overall fine for me :)

justoverclockl commented:
> I'm unable to read all of it thoroughly, but I skimmed through it and pushed some fixes, especially with accents that were broken in some files. I also changed the wording of some technical terms in the Contributing page, so that the original English terms are used (e.g. "branch" shouldn't be translated).
> 
> Overall fine for me :)

we waiting for @luceos

askvortsov1 commented on docs/.vuepress/config/locales/it/nav.js:
```diff
@@ -0,0 +1,13 @@
+module.exports = [
+    { text: 'Guide', link: '/it/' },
```
Should these be translated?

askvortsov1 replied:
The link texts I mean

justoverclockl replied:
link is usable even in italian

askvortsov1 commented on docs/.vuepress/config/locales/it/nav.js:
```diff
@@ -0,0 +1,13 @@
+module.exports = [
+    { text: 'Guide', link: '/it/' },
+    { text: 'Extend', link: '/extend/' },
```
Since extend docs are now translated, this should be switched back to '/it/extend/'

askvortsov1 commented on docs/.vuepress/config/locales/it/sidebar.js:
```diff
@@ -0,0 +1,104 @@
+module.exports = {
+  '/extend/': [
```
I think this needs to be '/it/extend/'

askvortsov1 commented on docs/it/code-of-conduct.md:
```diff
@@ -0,0 +1,51 @@
+- Se non utilizzi un tag riservato per l'uso multilingue, pubblica solo in italiano. Non possiamo aiutarti se non possiamo leggerti!
+- Ricorda, non è necessario firmare i tuoi post. Abbiamo il tuo profilo per farci sapere chi sei.
+
+Please make the effort to help us keep things organized. Time spent tidying up is time that we can't spend getting to know you, discussing your issues, and talking about Flarum. And that, after all, is what we're all here to do!
```
This is still missing

askvortsov1 commented on docs/it/extend/forms.md:
```diff
@@ -0,0 +1,114 @@
+# Forms and Requests
```
For pages under extend that aren't yet translated, please don't include the English source; instead, leave it blank and add text saying that translation is still in progress.

askvortsov1 reviewed (changes requested):
Found a few more things. Also, these pages still need the "this might get slightly outdated" disclaimer

justoverclockl commented:
> @matteocontrini and @Littlegolden thanks for reviewing, if the both of you can approve once everything is in order I think it's okay to merge 👍

for the component "outdated", i'm really confusing, i think i need some help

askvortsov1 commented:
I've unresolved a few more comments above that still need to be addressed.

For the "outdated" component, take a look at how this is implemented in the Chinese docs. If you're not comfortable setting that up, that's alright, we can add it in separately after merging the PR.

justoverclockl commented:
> I've unresolved a few more comments above that still need to be addressed.
> 
> For the "outdated" component, take a look at how this is implemented in the Chinese docs. If you're not comfortable setting that up, that's alright, we can add it in separately after merging the PR.

i think coul be merged now. all translation are ok except for the "outdated" componet, not comfortable with this

askvortsov1 commented on docs/.vuepress/config/locales/it/nav.js:
```diff
@@ -0,0 +1,13 @@
+module.exports = [
+    { text: 'Guida', link: '/it/' },
+    { text: 'Estendi', link: '/extend/' },
```
I'm pretty sure this needs to be `/it/extend`

askvortsov1 commented on docs/it/extend/update-b10.md:
```diff
@@ -0,0 +1,51 @@
+# Updating For Beta 10
```
If these aren't gonna be translated (which is fine, they're outdated and not essential) they should be removed.

askvortsov1 reviewed (changes requested):
Thank you so much for the collossal work done on this! A few last comments:

- I think I found one last broken link
- The upgrade guides can probably be completely removed

justoverclockl commented:
> Thank you so much for the collossal work done on this! A few last comments:
> 
> * I think I found one last broken link
> * The upgrade guides can probably be completely removed

its a pleasere, i really love your works, thats the minimum i can do to give you a little help

anyway i've removed (i hope) the useless file, and added correct path on nav.js

askvortsov1 commented on .gitignore:
```diff
@@ -1,4 +1,9 @@
-node_modules
\ No newline at end of file
+node_modules
+docs/it/extend/update-b8.md
```
These (gitignore) changes are unnecessary; also, links to the upgrade guides need to be removed from the navbar

justoverclockl replied:
how can i delete this file? (i'm new to github)

askvortsov1 replied:
No need to delete the files, just remove the added lines referencing .md files.

justoverclockl replied:
i think it's done!

askvortsov1 commented:
One last thing: could you translate the following text to italian?

> These documentation translations are community maintained, and might be outdated at times. Please refer to the English documentation for the latest version. If there are issues, please let us know [on github](https://github.com/flarum/docs/issues).

justoverclockl commented:
> One last thing: could you translate the following text to italian?
> 
> > These documentation translations are community maintained, and might be outdated at times. Please refer to the English documentation for the latest version. If there are issues, please let us know [on github](https://github.com/flarum/docs/issues).

Queste traduzioni della documentazione sono gestite dalla comunità e potrebbero essere obsolete. Fare riferimento alla documentazione in inglese per la versione più recente. Se riscontrate problemi, vi consigliamo di scriverci su GitHub.

justoverclockl commented:
> Thank you again so much for this collosal effort!

is the minimum that i can do, i will make another check for few little errors
